### PR TITLE
labels: Do not filter out app.kubernetes.io prefix

### LINKED ISSUE #8357: app.kubernetes.io labels are ignored
## Bug report

NetworkPolicy with labels selectors from the `app.kubernetes.io` prefix are ignored. Those labels are the recommended common labels from the [Kubernetes documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/).

Cilium docs doesn't seem to say anything about default label filtering as far as I can tell.

**General Information**

- Cilium version: 1.4.4 6506119be
- Kernel version: 4.19.37-3~bpo9+1
- Orchestration system version in use: kubernetes v1.14.1

**How to reproduce the issue**

1. Have a pod with a label `app.kubernetes.io/app=foo`
2. Try to enforce a policy matching on this label

**Additional informations**

This seems to be the case because of the `!io.kubernetes` filter in the default config. This is due to #880

https://github.com/cilium/cilium/blob/9a5e31fbd6274d277c457edabdc2470396bea16d/pkg/labels/filter.go#L161-L171

### PR DESCRIPTION
**Pull request by vadorovsky**
Prefix app.kubernetes.io is recommended for labels and annotations and
it shouldn't be considered as a private label used internally by
Kubernetes. Instead, kubernetes.io and *beta.kubernetes.io should be
filtered out.

Fixes #8357

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8574)

### CODE REVIEW

**Comment by mrostecki** [comment]
I would love to fix that problem just by using regex like `^.*(?<app.)kubernetes.io`, but unfortunately regexp library in Go doesn't seem to have any support for filtering out substrings...

**Comment by mrostecki** [comment]
test-me-please

**Comment by mrostecki** [comment]
test-me-please

**Comment by mrostecki** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]
Seems fine in principle to me, by the way here's the official docs link:

https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

I'll defer to k8s codeowners for the main review.

**Review by ianvernon — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by mrostecki** [comment]
test-me-please

**Comment by mrostecki** [comment]
Seems completely unrelated, but keeps showing up on Travis:

```
----------------------------------------------------------------------
FAIL: <autogenerated>:1: StoreConsulSuite.TestStoreOperations
store_test.go:204:
    c.Assert(expect(func() bool { return localKey2.deleted() == 0 }), IsNil)
... value *errors.errorString = &errors.errorString{s:"timeout while waiting for expected value"} ("timeout while waiting for expected value")
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
OOPS: 9 passed, 1 FAILED
--- FAIL: Test (130.87s)
FAIL
coverage: 10.1% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/kvstore/store	130.915s
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```

**Comment by jrajahalme** [comment]
Opened issue #8596 for the unit test flake.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
The unit test failure is unrelated and is a frequent flake in the unit tests (which is worrisome!).

**Comment by ianvernon** [comment]
@mrostecki can you please rebase so Travis can pick up the unit test fixes? After Travis succeeds, I can merge, since the CI passed.

**Comment by mrostecki** [comment]
test-me-please

**Pull request merged**
